server/static/upload.js

```diff
@@ -1,137 +1,242 @@
+        $('.add-sample-button').show();
+        $('.cancel-sample-button').hide();
+        
+        var fileTemplate = $('.sample-file-template').first().clone();
+        fileTemplate.attr('id', file.uniqueIdentifier);
+        fileTemplate.find('.file-name').html(file.fileName);
+        fileTemplate.find('.progress-bar').html('Ready...');
+        $('tbody[name="' + sampleName + '"]').append(fileTemplate);
+        fileTemplate.show();
+    
+    
+    });
+    r.on('fileProgress', function(file) {
+        var progress = getFileProgressElt(file);
+        var percent = Math.floor(file.progress() * 100);
+        progress.find('.progress-bar').html(percent + '%');
+        progress.find('.progress-bar').css({
+            width: percent + '%'
         });
-      r.on('uploadStart', function(){
-          // Show pause, hide resume
-          $('.progress-resume-link').hide();
-          $('.progress-pause-link').show();
-      });
+        progress.find('.progress-bar').css('color', 'white');
+        progress.find('.progress-bar').css('min-width', '2em')
+    });
+    r.on('uploadStart', function() {
+        //should probably hide options, or grey them out
+        //send message to server indicating token, samples, and files for db storage
+    });
+    r.on('fileSuccess', function(file, message) {
+        var progress = getFileProgressElt(file);
+        progress.find('.progress-bar').first()
+        .removeClass('progress-bar-striped active')
+        .addClass('progress-bar-success')
+        .html('Uploaded');
+    });
+    r.on('fileError', function(file, message) {
+        // Reflect that the file upload has resulted in error
+        var progress = getFileProgressElt(file);
+        var errorMsg = 'Error';
+        try {
+            errorMsg = $.parseJSON(message).message;
+        } catch (e) {
+        }
+        progress.find('.progress-bar')
+        .removeClass('progress-bar-striped active')
+        .addClass('progress-bar-danger')
+        .css({
+            width: '100%'
+        })
+        .html(errorMsg);
+    });
+    r.on('complete', function() {
+        //enable ability to add new samples/files? allow for upload again?
+        //populate the DCC table with files and empty the upload table?
+    });
+    
+    function getFileProgressElt(file) {
+        return $('#' + file.uniqueIdentifier);
     }
-
-    $('.resumable-drop').on('dragenter', function() {
-        $(this).addClass('resumable-dragover');
+    ;
+    
+    function getExtraParams(file, chunk) {
+        return {
+            'authToken': $('#auth-token').val()
+        };
+    }
+    ;
+    
+    // Remove file from resumable array using the file name
+    function removeResumableFile(fileName) {
+        fileObj = r.files.filter(function(obj) {
+            return obj.fileName == fileName
+        })[0];
+        fileIndex = r.files.indexOf(fileObj);
+        r.files.splice(fileIndex, 1);
```

Comment:
This seems bad. Would suggest reworking to use `.removeFile`, perhaps via `.getFromUniqueIdentifier`